extensions/reactive-datasource/runtime/src/main/java/io/quarkus/reactive/datasource/runtime/ThreadLocalPool.java

```diff
@@ -72,13 +72,19 @@ PoolType pool() {
     }
 
     private final void scanForAbandonedConnections() {
+        ArrayList<PoolAndThread> garbage = new ArrayList<>();
         for (PoolAndThread pair : allConnections) {
```

Comment:
This iteration may still fail if the allConnections list changes

Reply:
it's locked, it's not going to change at this time.